src/main/scala/demo/MiniDrawer.scala

```diff
@@ -0,0 +1,244 @@
+
+private val drawerWidth = 240
+
+private def openedMixin(theme: Theme) = new CSSObject {
+  width = drawerWidth
+  transition = theme.transitions_BaseTheme.create(
+    "width",
+    new Partial {
+      easing = theme.transitions_BaseTheme.easing.sharp
+      duration = theme.transitions_BaseTheme.duration.enteringScreen
+    }
+  )
+  overflowX = "hidden"
+}
+
+private def closedMixin(theme: Theme) =
+  new CSSObject {
+    transition = theme.transitions_BaseTheme.create(
+      "width",
+      new Partial {
+        easing = theme.transitions_BaseTheme.easing.sharp
+        duration = theme.transitions_BaseTheme.duration.leavingScreen
+      }
+    )
+    overflowX = "hidden"
+    width = s"calc(${theme.spacing(7)} + 1px)"
+
+    // todo: figure out how to apply this
+    // theme.breakpoints.up("sm")
+    //  [theme.breakpoints.up('sm')]: {
+    //  width: `calc(${theme.spacing(8)} + 1px)`,
+    // },
+  }
+
+val DrawerHeader = ScalaFnComponent.justChildren { c =>
+  div.styled
+    .fn { (theme, _) =>
+      new CSSObject {
+        display = "flex"
+        alignItems = "center"
+        justifyContent = "flex-end"
+        padding = theme.spacing(0, 1)
+      }
+        // necessary for content to be below app bar
+        .combineWith(theme.mixins_BaseTheme.toolbar)
+    }
+    .build()(c)
+}
+
+val AppBar = ScalaFnComponent.withChildren[Boolean] { (open, c) =>
+  mui.AppBar.styled
+    .fn((theme, _) =>
+      new CSSObject {
+        zIndex = theme.zIndex_BaseTheme.drawer + 1
+        transition = theme.transitions_BaseTheme.create(
+          js.Array("width", "margin"),
+          new Partial {
+            easing = theme.transitions_BaseTheme.easing.sharp
+            duration = theme.transitions_BaseTheme.duration.leavingScreen
+          }
+        )
+
+        if (open) {
+          marginLeft = drawerWidth
+          width = s"calc(100% - ${drawerWidth}px)"
+          transition = theme.transitions_BaseTheme.create(
+            js.Array("width", "margin"),
+            new Partial {
+              easing = theme.transitions_BaseTheme.easing.sharp
+              duration = theme.transitions_BaseTheme.duration.enteringScreen
+            }
+          )
+        }
+      }
+    )
+    .build()
+    .position("fixed")(c)
+}
+
+// todo: hack to set the css property
+trait CssDrawerPage extends CSSObject {
+  @unused
+  @JSName("& .MuiDrawer-paper")
+  var `MuiDrawer-paper`: js.UndefOr[Any | CSSObject] = js.undefined
+}
+
+val Drawer = ScalaFnComponent.withChildren[Boolean] { (open, c) =>
+  mui.Drawer.styled
+    .fn((theme, _) =>
+      new CssDrawerPage {
+        width = drawerWidth
+        flexShrink = 0
+        whiteSpace = "nowrap"
+        boxSizing = "border-box"
+        `MuiDrawer-paper` =
+          if (open) openedMixin(theme)
+          else closedMixin(theme)
+      }.combineWith(
+        if (open) openedMixin(theme)
+        else closedMixin(theme)
+      )
+    )
+    .build()
```

Comment:
save the result of `build` as a `val` and use that as a react component

Reply:
Got into an issue where we need to set some values based on some props. We found a workaround where we did an unsafe set and cast the value needed within the `fn` method.